runtime/lib/Handlers/api/utils.ts

```diff
@@ -43,90 +59,160 @@ export const getVariable = (path: string, data: any) => {
+        options.headers = { ...options.headers, ...formattedData.getHeaders() };
       }
-    });
-    options.headers = { ...options.headers, ...formData.getHeaders() };
-    options.data = formData;
-  } else if (bodyInputType === Node.Api.APIBodyType.URL_ENCODED) {
-    if (Array.isArray(body)) {
-      options.data = querystring.stringify(ReduceKeyValue(body));
-    } else {
-      options.data = querystring.stringify(body);
+    } else if (bodyInputType === 'urlEncoded') {
+      if (Array.isArray(body)) {
+        const tempObj: Record<string, any> = {};
+        body.forEach((b) => {
+          if (b.key) {
+            tempObj[b.key] = b.val;
+          }
+        });
+        formattedData = querystring.stringify(tempObj);
+      } else {
+        formattedData = querystring.stringify(body);
+      }
+      options.headers['Content-Type'] = 'application/x-www-form-urlencoded';
+    } else if (typeof body === 'string') {
+      formattedData = body;
+    } else if (bodyInputType === 'keyValue' || Array.isArray(body)) {
+      formattedData = {};
+      body.forEach((b) => {
+        if (b.key) {
+          formattedData[b.key] = b.val;
+        }
+      });
     }
-    options.headers['Content-Type'] = 'application/x-www-form-urlencoded';
-  } else if (typeof body === 'string') {
-    options.data = body;
-  } else if (bodyInputType === 'keyValue' || Array.isArray(body)) {
-    options.data = ReduceKeyValue(body);
+    options.data = formattedData;
   }
 
-  return options;
-};
-
-export const makeAPICall = async (nodeData: APINodeData) => {
-  const requestConfig = formatRequestConfig(nodeData);
+  options.validateStatus = () => true;
+  if (!options.headers['User-Agent']) {
+    options.headers['User-Agent'] = USER_AGENT;
+  }
 
-  const { data, headers, status } = await axios(requestConfig);
+  const response = await axios(options);
 
-  if (_.isObject(data) as any) {
-    data.VF_STATUS_CODE = status;
-    data.VF_HEADERS = headers;
+  if (_.isObject(response.data)) {
+    // @ts-expect-error assigned kvp thats not a part of type
+    response.data.VF_STATUS_CODE = response.status;
+    // @ts-expect-error assigned kvp thats not a part of type
+    response.data.VF_HEADERS = response.headers;
   }
 
-  const newVariables: Record<string, any> = {};
-  if (nodeData.mapping) {
-    nodeData.mapping.forEach((m) => {
-      if (m.var) {
-        newVariables[m.var] = getVariable(m.path, data);
-      }
-    });
+  return { data: response.data, headers: response.headers, status: response.status };
+};
+
+export const makeAPICall = async (data: APINodeData, runtime: Runtime) => {
+  try {
+    const response = await getResponse(data, runtime);
+    const newVariables: Record<string, any> = {};
+    if (data.mapping) {
+      data.mapping.forEach((m) => {
+        if (m.var) {
+          newVariables[m.var] = getVariable(m.path, response.data);
+        }
+      });
+    }
```

Comment:
Trying to utilize the array methods better.

```suggestion
    const newVariables = Object.fromEntries((data.mapping ?? [])
      .filter((map) => map.var)
      .map((map) => [map.var, getVariable(m.path, response.data)])
    );
```